### PR TITLE
[Sparse] Namespace migration and doc update

### PR DESCRIPTION
**Pull request by mufeili**
## Description

- Rename `python/mock_sparse2` to `python/sparse`
- Update examples and unit tests accordingly
- Re-direct the documentation to `python/sparse`, doc built and examined locally

We will also need to remove `mock_sparse` in another PR.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [ ] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by jermainewang** [comment]
@Rhett-Ying We need to update our doc server to enable the build of the new sparse lib.

File: docs/source/api/python/dgl.sparse_v0.rst [hunk 1]
```diff
@@ -1,86 +1,102 @@
...
 
-dgl.mock_sparse
+dgl.sparse
 =================================
 
-`dgl_sparse` is a library for sparse operators that are commonly used in GNN models.
+`dgl.sparse` is a library for sparse operators that are commonly used in GNN models.
 
 .. warning::
     This is an experimental package. The sparse operators provided in this library do not guarantee the same performance as their message-passing api counterparts.
```
Comment (jermainewang) [inline docs/source/api/python/dgl.sparse_v0.rst:9]: Should we remove this?

File: docs/source/api/python/dgl.sparse_v0.rst [hunk 2]
```diff
@@ -1,86 +1,102 @@
...
+.. currentmodule:: dgl.sparse
 
 .. class:: SparseMatrix
 
-    Class for creating a sparse matrix representation. The row and column indices of the sparse matrix can be the source
-    (row) and destination (column) indices of a homogeneous or heterogeneous graph.
+    Class for creating a sparse matrix representation
+
+    The row and column indices of the sparse matrix can be the destination
+    (row) and source (column) indices of a homogeneous graph.
```
Comment (jermainewang) [inline docs/source/api/python/dgl.sparse_v0.rst:20]: I don't think L19-L20 is needed.

File: docs/source/api/python/dgl.sparse_v0.rst [hunk 3]
```diff
@@ -1,86 +1,102 @@
...
 
     There are a few ways to create a sparse matrix:
 
     * In COO format using row and col indices, use :func:`create_from_coo`.
     * In CSR format using row pointers and col indices, use :func:`create_from_csr`.
     * In CSC format using col pointers and row indices, use :func:`create_from_csc`.
 
-    For example, we can create COO matrix as follows:
+    For example, we can create COO matrices as follows:
```
Comment (jermainewang) [inline docs/source/api/python/dgl.sparse_v0.rst:28]: Avoid using "we"

File: docs/source/api/python/dgl.sparse_v0.rst [hunk 4]
```diff
@@ -1,86 +1,102 @@
...
 
-    Case1: Sparse matrix with row and column indices without values.
+    Case1: Sparse matrix with row and column indices without values
 
-        >>> src = torch.tensor([1, 1, 2])
-        >>> dst = torch.tensor([2, 4, 3])
-        >>> A = create_from_coo(src, dst)
+        >>> dst = torch.tensor([1, 1, 2])
```
Comment (jermainewang) [inline docs/source/api/python/dgl.sparse_v0.rst:32]: Use row and col instead of src and dst.

**Review by jermainewang — APPROVED** [review]

File: docs/source/api/python/dgl.sparse_v0.rst (hunk 1 above)
Comment (mufeili) [reply docs/source/api/python/dgl.sparse_v0.rst:9]: I'm fine either way.

**Comment by Rhett-Ying** [comment]
> @Rhett-Ying We need to update our doc server to enable the build of the new sparse lib.

once this PR is merged into master, `dgl.mock_sparse` will be replaced by `dgl.sparse` in latest version, do we need additional operations?

**Comment by jermainewang** [comment]
> > @Rhett-Ying We need to update our doc server to enable the build of the new sparse lib.
> 
> once this PR is merged into master, `dgl.mock_sparse` will be replaced by `dgl.sparse` in latest version, do we need additional operations?

Yes, the building options/configs will be different too. There is a new build option `BUILD_SPARSE` need to turn on. CMake needs to be upgraded to 3.20+ too.

**Comment by Rhett-Ying** [comment]
@jermainewang doc server build is updated. pls take a look at here: https://docs.dgl.ai/en/latest/api/python/dgl.sparse_v0.html

**Pull request merged**